PreprocessingModule.py

```diff
@@ -0,0 +1,159 @@
+        link = None
+        with open(path_to_meta) as meta:
+            for line in meta:
+                if line.startswith("http"):
+                    link = line.rstrip("\n")
+                    continue
+                start, end, label = line.split(",")
+                start = datetime.strptime(start, '%H:%M:%S')
+                end = datetime.strptime(end, '%H:%M:%S')
+                start_sec = PreprocessingModule.__to_secs(start)
+                end_sec = PreprocessingModule.__to_secs(end)
+                duration = (end_sec - start_sec)
+                for _ in range(duration):
+                    bin_mask.append(1 if label == "song\n" else 0)
+        assert len(bin_mask) == end_sec, "Mask is not correct!"
+        return link, bin_mask
+
+    @staticmethod
+    def __to_secs(time):
+        total_seconds = time.second + time.minute * 60 + time.hour * 3600
+        return total_seconds
+
+    @staticmethod
+    def preprocess_train(path_to_meta, path_to_video, path_to_audio, seq_len=100):
+        """
+        This method is to be used when training the model.
+        Takes the video either from the given location or downloads using the link, converts it to audio,
+        cuts audio in pieces, translates the segments-info into a binary mask. Creates a pkl file with paths
+        to all audio-cuts and the binary-mask.
+
+        :param path_to_meta: path to meta-info about the audio(video);
+        :param path_to_video: path to the existing video to be converted to training data
+                              (if it doesn't exist can be downloaded from YouTube using the
+                              link given in 1st line of meta-info file);
+        :param path_to_audio: path to where audio is to be placed after conversion;
+        :param seq_len: the length of sample in seconds to which the audio is cut;
+        :return: void.
+        """
+        assert os.path.isfile(path_to_meta), "Meta-info file {} not found".format(path_to_meta)
+
+        link, bin_mask = PreprocessingModule.__preprocess_meta(path_to_meta)
+        if link and not os.path.isfile(path_to_video):
+            PreprocessingModule.__download_from_youtube(link, path_to_video)
+        PreprocessingModule.convert_to_wav(path_to_video, path_to_audio)
+        files_list = PreprocessingModule.__cut_in_pieces(path_to_audio, seq_len)
+        PreprocessingModule.__generate_pkl(files_list, bin_mask)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Preprocessing')
+
+    parser.add_argument('-i', action="store", dest="mode")
+    parser.add_argument('-v', action="store", dest="path_to_video")
+    parser.add_argument('-a', action="store", dest="path_to_audio")
+    parser.add_argument('-m', action="store", dest="path_to_meta")
+    parser.add_argument('-s', action="store", dest="seq_len")
+
+    args = parser.parse_args()
+    if args.seq_len is None:
+        args.seq_len = 100
+
+    if args.mode == "train":

```

Comment:
I think it's worth specify both "train" and "cut" modes (not only train) and also add information about these modes in --help of the script